### ISSUE TITLE
Separate import path for headless components and icons

### ISSUE THREAD

**Issue by Emil-Hammer** [vidstack/vidstack#841]
### Current Behavior:
Installing Vidstack with the intention of using it headless still throws a warning about missing peerDependencies.

### Expected Behavior:
Installing Vidstack doesn't require installing additional peerDependencies especially not media-icons.

### Environment:
"vidstack": "0.5.6"
"@vidstack/react": "^0.5.6"

### Anything Else?
I can't see the reason for maverick.js being a peerDependency since Vidstack is very restrictive around the versions I can use. Furthermore I'm not using it as a "plugin" for Vidstack. Only internal Vidstack code is using it. For media-icons it makes more sense for me that it is split away from this package.

I might be missing something so feel free to correct me. This is just not the usual behaviour I've seen from other packages.

**Comment by mihar-22** [comment]
Thanks for bringing these issues up @Emil-Hammer!

1. **Headless Components**

There are plans for the upcoming release to split out our headless package separately and actually render nothing inside our components. Today it's done a little hacky with CSS and we realize it's not right. This change will also require slightly re-working our React components to feel more like existing libraries such as Radix (e.g., use render props for dynamic parts such as time slider chapters).

Imports for React might end up something like:

```ts
import { PlayButton } from '@vidstack/react/headless'
```

2. **Maverick**

You're right here. It was originally listed as a dep to avoid bundling Maverick multiple times for future libraries (outside of Vidstack). For the current time since we are version locked, and we don't have plans just yet of using Maverick outside of Vidstack, it does make sense to bundle it. This will be done for the upcoming release.

3. **Media Icons**

I've been thinking about this too and I think we'll split icons out from the main bundle in the next release. This will mean `media-icons` will only be required when using the icons from the following import paths:

```ts
// HTML
import 'vidstack/define/media-icon';

// React
import {  PlayIcon, PauseIcon, MuteIcon, ... } from '@vidstack/react/icons';
```

**Comment by mihar-22** [comment]
This issue is outdated now and is resolved since our new [`1.0@next` release](https://github.com/vidstack/vidstack/discussions/935). If the issue persits after upgrading, feel free to let me know here and we can reopen. If the issue has changed then please open a new one.
